libecl/src/ecl_file_view.c

```diff
@@ -0,0 +1,785 @@
+  return ecl_file_kw_get_type( file_kw );
+}
+
+int ecl_file_view_iget_named_size( const ecl_file_view_type * ecl_file_view , const char * kw , int ith) {
+  ecl_file_kw_type * file_kw = ecl_file_view_iget_named_file_kw( ecl_file_view , kw , ith );
+  return ecl_file_kw_get_size( file_kw );
+}
+
+
+void ecl_file_view_replace_kw( ecl_file_view_type * ecl_file_view , ecl_kw_type * old_kw , ecl_kw_type * new_kw , bool insert_copy) {
+  int index = 0;
+  while (index < vector_get_size( ecl_file_view->kw_list )) {
+    ecl_file_kw_type * ikw = vector_iget( ecl_file_view->kw_list , index );
+    if (ecl_file_kw_ptr_eq( ikw , old_kw)) {
+      /*
+         Found it; observe that the vector_iset() function will
+         automatically invoke the destructor on the old_kw.
+      */
+      ecl_kw_type * insert_kw = new_kw;
+
+      if (insert_copy)
+        insert_kw = ecl_kw_alloc_copy( new_kw );
+      ecl_file_kw_replace_kw( ikw , ecl_file_view->fortio , insert_kw );
+
+      ecl_file_view_make_index( ecl_file_view );
+      return;
+    }
+    index++;
+  }
+  util_abort("%s: could not find ecl_kw ptr: %p \n",__func__ , old_kw);
+}
+
+
+bool ecl_file_view_load_all( ecl_file_view_type * ecl_file_view ) {
+  bool loadOK = false;
+
+  if (fortio_assert_stream_open( ecl_file_view->fortio )) {
+    int index;
+    for (index = 0; index < vector_get_size( ecl_file_view->kw_list); index++) {
+      ecl_file_kw_type * ikw = vector_iget( ecl_file_view->kw_list , index );
+      ecl_file_kw_get_kw( ikw , ecl_file_view->fortio , ecl_file_view->inv_map);
+    }
+    loadOK = true;
+  }
+
+  if (ecl_file_view_flags_set( ecl_file_view , ECL_FILE_CLOSE_STREAM))
+    fortio_fclose_stream( ecl_file_view->fortio );
+
+  return loadOK;
+}
+
+
+/*****************************************************************/
+
+
+
+void ecl_file_view_add_kw( ecl_file_view_type * ecl_file_view , ecl_file_kw_type * file_kw) {
+  if (ecl_file_view->owner)
+    vector_append_owned_ref( ecl_file_view->kw_list , file_kw , ecl_file_kw_free__ );
+  else
+    vector_append_ref( ecl_file_view->kw_list , file_kw);
+}
+
+void ecl_file_view_free( ecl_file_view_type * ecl_file_view ) {
+  vector_free( ecl_file_view->child_list );
+  hash_free( ecl_file_view->kw_index );
+  stringlist_free( ecl_file_view->distinct_kw );
+  vector_free( ecl_file_view->kw_list );
+  free( ecl_file_view );
+}
+
+void ecl_file_view_free__( void * arg ) {
+  ecl_file_view_type * ecl_file_view = ( ecl_file_view_type * ) arg;
+  ecl_file_view_free( ecl_file_view );
+}
+
+
+int ecl_file_view_get_num_named_kw(const ecl_file_view_type * ecl_file_view , const char * kw) {
+  if (hash_has_key(ecl_file_view->kw_index , kw)) {
+    const int_vector_type * index_vector = hash_get(ecl_file_view->kw_index , kw);
+    return int_vector_size( index_vector );
+  } else
+    return 0;
+}
+
+void ecl_file_view_fwrite( const ecl_file_view_type * ecl_file_view , fortio_type * target , int offset) {
```

Comment:
Writing to fortio is the only possibility, so I thought it was not necessary to include it as part of the name.